src/lib/device-state.ts

```diff
@@ -72,25 +82,32 @@ export const releaseFromApp = (
 				},
 			},
 		})
-		.then(([release]: AnyObject[]) => release);
+		.then(([release]: Release[]) => release);
 };
 
 export const serviceInstallFromImage = (
-	device: AnyObject,
-	image?: AnyObject,
-): undefined | AnyObject => {
+	device: Device,
+	image?:
+		| Image
+		| (Pick<Image, Exclude<keyof Image, 'is_a_build_of__service'>> & {
+				is_a_build_of__service: number;
+		  }),
+): ServiceInstall | undefined => {
 	if (image == null) {
 		return;
 	}
 
 	let id: number;
 	if (_.isObject(image.is_a_build_of__service)) {
-		id = image.is_a_build_of__service.__id;
+		id = (image.is_a_build_of__service as PineDeferred).__id;
```

Comment:
For such cases, in the UI we use a `getId` method that works both on PineDeferred and expanded resources. I can bring this in here as well if you also like it:
```
export const getId = (obj: BalenaSdk.NavigationResource) =>
	getExpandedId(obj) || getDeferredId(obj);

export const getExpanded = <T>(obj: BalenaSdk.NavigationResource<T>) =>
	(_.isArray(obj) && obj[0]) || undefined;

export const getExpandedProp = <T, K extends keyof T>(
	obj: BalenaSdk.NavigationResource<T>,
	key: K,
) => (_.isArray(obj) && obj[0] && obj[0][key]) || undefined;

export const getExpandedId = (obj: BalenaSdk.NavigationResource) =>
	getExpandedProp(obj, 'id');

// custom type guard function
export const isPineDeferred = (obj: any): obj is BalenaSdk.PineDeferred =>
	obj && '__id' in obj;

// function Typing overloads
export function getDeferredId(obj: BalenaSdk.PineDeferred): number;
export function getDeferredId(
	obj: BalenaSdk.NavigationResource,
): number | undefined;
export function getDeferredId(obj: BalenaSdk.NavigationResource) {
	return (isPineDeferred(obj) && obj.__id) || undefined;
}
```